src/axom/inlet/tests/inlet_errors.cpp

```diff
@@ -0,0 +1,350 @@
+// Copyright (c) 2017-2021, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "gtest/gtest.h"
+
+#include "axom/inlet/Inlet.hpp"
+#include "axom/inlet/tests/inlet_test_utils.hpp"
+
+using axom::inlet::Inlet;
+using axom::inlet::VerificationError;
+using axom::sidre::DataStore;
+
+template <typename InletReader>
+Inlet createBasicInlet(const std::string& luaString)
+{
+  std::unique_ptr<InletReader> reader(new InletReader());
+  reader->parseString(axom::inlet::detail::fromLuaTo<InletReader>(luaString));
+  const bool enableDocs = true;
+  return Inlet(std::move(reader), enableDocs);
+}
+
+template <typename InletReader>
+class inlet_errors : public ::testing::Test
+{ };
+
+TYPED_TEST_SUITE(inlet_errors, axom::inlet::detail::ReaderTypes);
+
+TYPED_TEST(inlet_errors, required_field)
+{
+  std::string testString = "";
+  Inlet inlet = createBasicInlet<TypeParam>(testString);
+
+  inlet.addInt("foo", "foo's description").required();
+
+  std::vector<VerificationError> errors;
+  EXPECT_FALSE(inlet.verify(&errors));
+  // Need something about "foo" and "Required"
+  EXPECT_TRUE(
+    std::any_of(errors.begin(), errors.end(), [](const VerificationError& err) {
+      return (static_cast<std::string>(err.path) == "foo") &&
+        (err.message.find("Required") != std::string::npos);
+    }));
+}
+
+TYPED_TEST(inlet_errors, wrong_type)
+{
+  std::string testString = "foo = 'hello'";
+  Inlet inlet = createBasicInlet<TypeParam>(testString);
+
+  inlet.addInt("foo", "foo's description");
+
+  std::vector<VerificationError> errors;
+  EXPECT_FALSE(inlet.verify(&errors));
+  // Need something about "foo" and "wrong type"
+  EXPECT_TRUE(
```

Comment:
It would be nice to provide something like the below to simplify verifying expected errors:
```cpp
ASSERT_CONTAINS_ERROR(errors, [](const VerificationError &err){...});
// with gmock
EXEPCT_THAT(error, ContainsErrorMatching([](const VerificationError &err));
```

I don't know where would be a good place to put it though, such that both other Axom libraries and non-Axom users of Inlet could access it. I don't know if you could to do a good chunk of it in Inlet itself and leave the other libraries/users to define a small bit of gmock glue code themselves.

Reply:
Does axom build with gmock enabled?
I don't recall us using it elsewhere in the code (and I personally have not used it).

@estebanpauli -- are you using it w/ klee?

Reply:
The syntax looks really clean